### PR TITLE
Fix PMD violations and rename method, issue #744

### PR DESCRIPTION
**Pull request by mkordas**

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java [hunk 1]
```diff
@@ -214,16 +214,13 @@ public void visitToken(DetailAST ast)
...
         final String literal = ast.getText();
 
-        if (hasUnicodeChar(literal)) {
-            if (!(allowByTailComment && haastrailComment(ast)
-                    || isAllCharactersEscaped(literal)
-                    || allowEscapesForControlCharacters
-                            && isOnlyUnicodeValidChars(literal, sUnicodeControl)
-                    || allowNonPrintableEscapes
-                            && isOnlyUnicodeValidChars(literal, sNonPrintableChars)))
-            {
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java]: we had strange formatting here...

**Comment by romani** [comment]
that formatting is intentional, to show that && is not on the same level of execution as ||.
It is kind a way to not put extra "(" and ")" and in the same time show structure.

we will try to address that during https://github.com/checkstyle/checkstyle/issues/270 .

Formatting that was before is more readable and clear. As we want to remove braces we need to format code to show execetution structure, in other case braces should stay.

**Comment by mkordas** [comment]
@romani, I totally agree, however such indentation is very fragile - IDEs do not support it.
I'll prepare a fix soon.

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**